src/gui/qgstemporalcontrollerdockwidget.cpp

```diff
@@ -0,0 +1,238 @@
+  : QgsDockWidget( parent )
+{
+  setupUi( this );
+  setWindowTitle( name );
+
+  mNavigationObject = new QgsTemporalNavigationObject( this );
+  settingsDialog();
+
+  connect( mForwardButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::forwardButton_clicked );
+  connect( mBackButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::backButton_clicked );
+  connect( mNextButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::nextButton_clicked );
+  connect( mPreviousButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::previousButton_clicked );
+  connect( mStopButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::stopButton_clicked );
+
+  connect( mStartDateTime, &QDateTimeEdit::dateTimeChanged, this, &QgsTemporalControllerDockWidget::updateTemporalExtent );
+  connect( mEndDateTime, &QDateTimeEdit::dateTimeChanged, this, &QgsTemporalControllerDockWidget::updateTemporalExtent );
+  connect( mSpinBox, qgis::overload<int>::of( &QSpinBox::valueChanged ), this, &QgsTemporalControllerDockWidget::updateFrameDuration );
+  connect( mTimeStepsComboBox, qgis::overload<int>::of( &QComboBox::currentIndexChanged ), this, &QgsTemporalControllerDockWidget::updateFrameDuration );
+  connect( mSlider, &QSlider::valueChanged, this, &QgsTemporalControllerDockWidget::timeSlider_valueChanged );
+
+  connect( mNavigationObject, &QgsTemporalNavigationObject::updateTemporalRange, this, &QgsTemporalControllerDockWidget::updateSlider );
+
+  connect( mSettingsDialog->mapSettingsWidget(), &QgsTemporalMapSettingsWidget::frameRateChanged, this, &QgsTemporalControllerDockWidget::updateFrameRate );
+
+  connect( mSettings, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::settings_clicked );
+  connect( mSetToProjectTimeButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::setDatesToProjectTime );
+
+  init();
+}
+
+QgsTemporalControllerDockWidget::~QgsTemporalControllerDockWidget()
+{
+}
+
+void QgsTemporalControllerDockWidget::init()
+{
+  QgsDateTimeRange range;
+
+  if ( QgsProject::instance()->timeSettings() )
+    range = QgsProject::instance()->timeSettings()->temporalRange();
+  QLocale locale;
+
+  mStartDateTime->setDisplayFormat( locale.dateTimeFormat() );
+  mEndDateTime->setDisplayFormat( locale.dateTimeFormat() );
+
+  if ( range.begin().isValid() && range.end().isValid() )
+  {
+    mStartDateTime->setDateTime( range.begin() );
+    mEndDateTime->setDateTime( range.end() );
+  }
+
+  mSetToProjectTimeButton->setToolTip( tr( "Set to project time" ) );
+
+  QStringList listSteps = ( QStringList() << tr( "Seconds" ) << tr( "Minutes" ) << tr( "Hours" )
+                            << tr( "Days" ) <<
+                            tr( "Months" ) << ( "Years" ) );
+
+  mTimeStepsComboBox->addItems( listSteps );
+  mTimeStepsComboBox->setCurrentIndex( 2 );
+
+  mSpinBox->setMinimum( 1 );
```

Comment:
This should be double spin box, and the minimum should be something like 0.0000001

You may want to advance by 0.5 seconds or less in some maps